website/content/specification/syntax/constraints.md

```diff
@@ -179,6 +179,16 @@ A constraint may have an OPTIONAL [`@level`](#level) attribute and/or an OPTIONA
 
 If defined, the `<message>` value MUST be a [Metaschema string value](/specification/datatypes#string). It MAY contain a Metapath expression templates that starts with `{`, contains a Metapath expression, and ends with `}`.  When evaluating a template Metapath expression, the context of the Metapath [evaluation focus](#constraint-processing) MUST be the failing value node.
 
+##`matches`
+
+The `matches` constraint is a type of Metaschema constraint that restricts field or flag value(s) based on a component matching the target Metapath expression. Each one of these are discussed below.
+
+A match can be made by 2 different ways based on `@datatypes` and based on a regular expression. 
+
+The `@target` attribute of an `matches` constraint specifies the node in a component definition whose value is equal. The `matches` constraint MUST define a [`target`](#target) with a Metapath expression. The node MUST also match on `@datatype`. 
+
+The `@target` attribute of a `matches` constraint can be matched based on its value and a regular expression that the node MUST match on.
```

Comment:
Can we make sure in the beginning we call it `@regex` so we know the syntax and it is an attribute on matches and not a child element?

What kind of regular expression is it? There are a few flavors in different programming languages and runtimes that have their own syntax. Which kind do we use in Metaschema?

You write you MUST match the syntax pattern. But is this attribute optional or mandatory in all cases? If it isn't there and you MUST match on it, how would that work?

Reply:
[Regular Expressions](https://pages.nist.gov/metaschema/specification/datatypes/#regular-expressions-in-schemas)

Reply:
I believe this a single line matching regular expression.

The `@target` attribute is optional.

Reply:
Arminta covered this in https://github.com/usnistgov/metaschema/pull/525/commits/34ce6424a492f28c77334e14b93b53b96620d160. Looks good now.